contracts/RevisedPixelmonTrainerAdventure/PxChainlinkManager.sol

```diff
@@ -0,0 +1,181 @@
+    mapping(uint256 => WeeklyRandomNumber) weeklyRandomNumbers;
+
+    /// @notice address of trainer adventure smart contract
+    address public trainerAdventureContractAddress;
+
+    // @notice The maximum gas price to pay for a request to Chainlink in wei.
+    bytes32 public keyHash;
+    /// @notice How many confirmations the Chainlink node should wait before responding
+    uint16 requestConfirmations = 3;
+    /// @notice Chainlink subscription ID that used for sending request
+    uint64 public chainLinkSubscriptionId;
+    /// @notice Gas limit used to call Chainlink
+    uint32 public callbackGasLimit = 400000;
+    /// @notice Address that is able to call Chainlink
+    VRFCoordinatorV2Interface internal COORDINATOR;
+    /// @notice Last request ID to Chainlink
+    uint256 public lastRequestId;
+    /// @notice Collection of chainink request ID
+    uint256[] public requestIds;
+    /// @notice Map of request to Chainlink
+    mapping(uint256 => Request) public requests;
+
+    /// @notice Emit when calling fulfillRandomWords function
+    /// @param weekNumber The week number when the request is sent to Chainlink
+    /// @param RandomWords The input random words
+    event ChainlinkRandomNumberSet(uint256 weekNumber, uint256[] RandomWords);
+
+    /// @notice Constructor function
+    /// @param _vrfCoordinator The address of the Chainlink VRF Coordinator contract
+    /// @param _chainLinkSubscriptionId The Chainlink Subscription ID that is funded to use VRF
+    /// @param _keyHash The gas lane to use, which specifies the maximum gas price to bump to.
+    ///        More https://docs.chain.link/docs/vrf/v2/subscription/supported-networks/#configurations
+    constructor(
+        address _signer,
+        address _vrfCoordinator,
+        uint64 _chainLinkSubscriptionId,
+        bytes32 _keyHash
+    ) VRFConsumerBaseV2(_vrfCoordinator) EIP712(SIGNING_DOMAIN, SIGNATURE_VERSION) {
+        COORDINATOR = VRFCoordinatorV2Interface(_vrfCoordinator);
+        keyHash = _keyHash;
+        chainLinkSubscriptionId = _chainLinkSubscriptionId;
+        SIGNER = _signer;
+    }
+
+    /// @notice Sets Signer wallet address
+    /// @dev This function can only be executed by the contract owner
+    /// @param signer Signer wallet address for signature verifition
+    function setSignerAddress(address signer) external onlyOwner {
+        SIGNER = signer;
+    }
+
+    /// @notice Recovers signer wallet from signature
+    /// @dev View function for signature recovering
+    /// @param weekNumber Week number for claim
+    /// @param claimIndex Claim index for a perticular user for a week
+    /// @param walletAddress Token owner wallet address
+    /// @param signature Signature from signer wallet
+    function recoverSignerFromSignature(
+        uint256 weekNumber,
+        uint256 claimIndex,
+        address walletAddress,
+        bytes calldata signature
+    ) external view returns (bool) {
```

Comment:
Have we unit tested this with invalid signature strings?

Reply:
yes